<?xml version="1.0"?>
<!--|
    Copyright Inria July 2017
    This project has received funding from the European Research Council (ERC) under
    the European Union's Horizon 2020 research and innovation programme (grant
    agreement No 637972) - see http://www.resibots.eu

    Contributor(s):
      - Matthias Mayr (matthias.mayr@cs.lth.se)
      - Konstantinos Chatzilygeroudis (konstantinos.chatzilygeroudis@inria.fr)
      - Rituraj Kaushik (rituraj.kaushik@inria.fr)
      - Roberto Rama (bertoski@gmail.com)


    This software is governed by the CeCILL-C license under French law and
    abiding by the rules of distribution of free software.  You can  use,
    modify and/ or redistribute the software under the terms of the CeCILL-C
    license as circulated by CEA, CNRS and INRIA at the following URL
    "http://www.cecill.info".

    As a counterpart to the access to the source code and  rights to copy,
    modify and redistribute granted by the license, users are provided only
    with a limited warranty  and the software's author,  the holder of the
    economic rights,  and the successive licensors  have only  limited
    liability.

    In this respect, the user's attention is drawn to the risks associated
    with loading,  using,  modifying and/or developing or reproducing the
    software by the user in light of its specific status of free software,
    that may mean  that it is complicated to manipulate,  and  that  also
    therefore means  that it is reserved for developers  and  experienced
    professionals having in-depth computer knowledge. Users are therefore
    encouraged to load and test the software's suitability as regards their
    requirements in conditions enabling the security of their systems and/or
    data to be ensured and,  more generally, to use and operate it in the
    same conditions as regards security.

    The fact that you are presently reading this means that you have had
    knowledge of the CeCILL-C license and that you accept its terms.

|-->
<launch>
    <arg name="worker_nr" default="1"/>
    <arg name="robot_base_name" default="bh_robot"/>
    <arg name="libraries_list" default="[skiros2_moveit_lib]"/>
    <arg name="primitive_list" default="[
            plan_cartesian_fast]"/>
    <arg name="skill_list" default="[
        execute_trajectory,
        goto]"/>
    <arg name="robot_ontology_prefix" default="skiros" />
    <arg name="deploy" default="true" />
    <arg name="verbose" default="true"/>
    <arg name="debug" default="false"/>

    <include file="$(find skiros2)/launch/skill_mgr.launch">
        <arg name="verbose" value="$(arg verbose)"/>
        <arg name="debug" value="$(arg debug)"/>
        <arg name="libraries_list" value="$(arg libraries_list)"/>
        <arg name="primitive_list" value="$(arg primitive_list)"/>
        <arg name="skill_list" value="$(arg skill_list)" />
        <arg name="robot_ontology_prefix" value="$(arg robot_ontology_prefix)"/>
        <!-- For testing the skill integration -->
        <!-- <arg name="robot_name" value="test_robot"/> -->
        <arg name="robot_name" value="$(arg robot_base_name)_$(arg worker_nr)"/>
        <arg name="deploy" value="$(arg deploy)"/>
    </include>

</launch>